runtime/commonTest/src/kotlinx/serialization/json/JsonTestBase.kt

```diff
@@ -133,4 +134,9 @@ abstract class JsonTestBase {
             assertEquals(data, deserialized)
         }
     }
+
+    inline fun <reified T : Throwable> assertFailsWith(message: String, block: () -> Unit) {
```

Comment:
I suggest renaming to `assertFailsWithMessage` because stdlib has very similar overload (https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-fails-with.html#assertfailswith), but `message` there is message used in test assertion.